lib/foreman_maintain/cli/report_command.rb

```diff
@@ -0,0 +1,25 @@
+module ForemanMaintain
+  module Cli
+    class ReportCommand < Base
+      extend Concerns::Finders
+
+      option '--output', 'FILE', 'Output the generate report into FILE'
+      subcommand 'generate', 'Generates the usage reports' do
+        def execute
+          scenario = run_scenario(Scenarios::Report::Generate.new({}, [:reports])).first
+
+          # description can be used too
+          report_data = scenario.steps.map(&:data).compact.reduce(&:merge).transform_keys(&:to_s)
```

Comment:
does that take all reports, their `.data` and merge those hashes? what if two reports accidentally define the same hash key?

Reply:
Yes. I think that's very unlikely as the keys are very descriptive. Is it worth printing a warning to STDERR? Or perhaps better, having a some sort of a test that check for all keys in all checks? That may be a bit harder though as some of them are generated dynamically. Perhaps each check could have some explicit prefix that could that be verified, but that's likely schema v2 thing.

Reply:
What if we add an `add_data(key, value)` method to `Report`, and it does something like
```ruby
data[self.class.name+'_'+key] = value
```

I wouldn't block on this, but I think it's a potential source for hard to debug issues once a "not so descriptive" name leaks into the set.

Reply:
@adamruzicka @ShimShtein thoughts?

Reply:
It would also save the `transform_keys(to_s)` here, which makes the code more readable?

Reply:
I'd leave this for v2. While we're at it, we should maybe shuffle things around a bit, the report classes are currently named somewhat arbitrarily

Reply:
If you keep something big like this I would find it more readable to split it across lines:

```
          report_data = scenario.
            steps.
            map(&:data).
            compact.
            reduce(&:merge).
            transform_keys(&:to_s)
```